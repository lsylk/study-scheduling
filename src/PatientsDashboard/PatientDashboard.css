.Patient-form {
    display: flex;
    flex-wrap: wrap;
}

.Patient-form > span {
    margin: 10px;
    display: inline-block;
}