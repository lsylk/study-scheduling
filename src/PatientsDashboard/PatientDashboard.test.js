import React from "react";

import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import PatientDashboard from "./PatientsDashboard.tsx";

import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';

import generalReducer from '../store/reducers/general';
import patientReducer from '../store/reducers/patient';
import procedureReducer from '../store/reducers/procedure';
import doctorReducer from '../store/reducers/doctor';
import roomReducer from '../store/reducers/room';

const rootReducer = combineReducers({
    tab: generalReducer,
    patients: patientReducer,
    procedures: procedureReducer,
    doctors: doctorReducer,
    rooms: roomReducer
});
const store = createStore(rootReducer);

configure({ adapter: new Adapter() });
describe("PatientDashboard component", () => {
  const wrapper = mount(<Provider store={store}><PatientDashboard /></Provider>);

  it("renders PatientDashboard component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders New Patient Form", () => {
    expect(
      wrapper
        .find("h4")
        .at(0)
        .text()
    ).toContain("New Patient Form");
  });

  it("renders New Patient Form element", () => {
    expect(wrapper.find("form")).toHaveLength(1);
  });

  it("renders Patient List", () => {
    expect(
      wrapper
        .find("h4")
        .at(1)
        .text()
    ).toContain("Patient List");
  });

  it("renders Patient Table element", () => {
    expect(wrapper.find("table")).toHaveLength(1);
  });
});