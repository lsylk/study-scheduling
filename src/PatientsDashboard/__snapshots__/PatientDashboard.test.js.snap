// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientDashboard component renders PatientDashboard component 1`] = `ReactWrapper {}`;
